src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/HttpContentJsonExtensions.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Net.Mime;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public static class HttpContentJsonExtensions
+    {
+        public static Task<object?> ReadFromJsonAsync(this HttpContent content, Type type, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore(content, type, options, cancellationToken);
+        }
+
+        public static Task<T> ReadFromJsonAsync<T>(this HttpContent content, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            return ReadFromJsonAsyncCore<T>(content, options, cancellationToken);
+        }
+
+        private static async Task<object?> ReadFromJsonAsyncCore(HttpContent content, Type type, JsonSerializerOptions? options, CancellationToken cancellationToken)
+        {
+            Stream contentStream = await GetJsonStreamFromContentAsync(content, cancellationToken).ConfigureAwait(false);
+            return await JsonSerializer.DeserializeAsync(contentStream, type, options, cancellationToken);
```

Comment:
.ConfigureAwait(false)

Reply:
We have an analyzer that flags this.  Is this not producing a warning-as-error in the build?

Reply:
Interesting.  I did notice in a different project yesterday that CA2007 was being flagged as warn as error.  Something seems wrong with the CI state of this PR /cc @safern, only 9 checks.

Reply:
@joperezr just triggered the build and I'm following up with FR on the issue.

Reply:
Ah... DeserializeAsync returns `ValueTask<T>`, and: https://github.com/dotnet/roslyn-analyzers/issues/2700

Reply:
No everything makes sense.

Reply:
This is a `ValueTask`, that might be why the analyzer does not flag it.